### ISSUE TITLE
[Bug] Double vehicle spawn and unsync

### ISSUE THREAD

**Issue by Desoroxxx** [Starystars67/BeamNG-MP#443]
**Have you read our FAQ/Wiki/#before-you-ask ?**
Yes

**Describe the bug**
When I join a server for the first time I get a lua error after spawning my vehicle additionally a second vehicle spawns, and I am not synced to anybody else, I can see everyone move and do their life but no one can see me move, other player only see one vehicle

**Screenshots**
![image](https://github.com/BeamMP/BeamMP/assets/82710983/349f23e2-d3e8-4710-98c8-a3fa4e03b673)

**Desktop (please complete the following information):**
 - OS + Version: Windows 11

**Additional context**
Once the double vehicles are spawned I cannot spawn or replace them, my camera just moves to someone else vehicle

This does not happen after I quit the server and join again

**Comment by OfficialLambdax** [comment]
line 876 is `players[vehicle.ownerID]:addVehicle(vehicle)`
players table definition is [here](https://beammp.github.io/BeamMP/modules/MPVehicleGE.html#players)
players table is indexed by the player id. here is the ownerID of a vehicle entry read and used as the index. The ownerid equals to the playerid of the spawning player, according to [definition](https://beammp.github.io/BeamMP/modules/MPVehicleGE.html#vehicles_subtable)

We cannot see this value in the image of the init post. However since OP states that they have spawned the vehicle themself and because the image contains `"pid":-1`, owner id probably is also -1. and indexing a table with -1 is invalid.

-1 is not even a valid player id. But -1 is the default playerid within [MPConfig.lua](https://github.com/BeamMP/BeamMP/blob/development/lua/ge/extensions/MPConfig.lua#L18), which is overwritten by the launcher as the last step of the [sync](https://github.com/BeamMP/BeamMP/blob/development/lua/ge/extensions/MPGameNetwork.lua#L269). This packet is send from the launcher to the game directly as soon as the game connects to the proxy socket (default: 4445) of the launcher.

In other words, this is a sympton of a improper syncing process or improper connection between the game and the launchers proxy socket. Or maybe this is tied to the sometimes malfunctioning packet splits that can occur with the socket lib

**Comment by Desoroxxx** [comment]
I would need to try this again on another server than CaRP just to be sure but I do not think this happens anymore, can't recall because didn't pay attention but I think it got fix with some BeamNG hotfix or BeamMP update

**Comment by xerxes-at** [comment]
Hey,
I got the same symptoms but get them all the time.
In my case it is caused by a 3rd party tool called Proxifier, which I assume creates some issue for the TCP socket communication.
Is there anything I can collect and share to help resolve the issue?

**Comment by lionkor** [comment]
Where, how and why are you using proxifier?

**Comment by xerxes-at** [comment]
I use it on my PC (not on my game server) to send traffic of certain applications (mostly Firefox) through (paid) proxy servers so I don't have to use a VPN all the time and so I can apply some logic to which traffic is sent where. If I don't set-up rules all traffic of an application is sent directly, which is the case for BeamMP. Proxifier however does register a Driver which is used to intercept the connections.
